ml/src/main/scala/ai/h2o/sparkling/ml/features/H2OWord2VecTokenizer.scala

```diff
@@ -39,28 +39,30 @@ class H2OWord2VecTokenizer(override val uid: String)
   @DeveloperApi
   override def transformSchema(schema: StructType): StructType = {
     require(getInputCol != null, "Input column has to be specified!")
-    StructType(schema.fields ++ Array(StructField(getOutputCol, StringType)))
+    StructType(schema.fields ++ Array(StructField(getOutputCol(), StringType)))
   }
 
-  private val MARKER = "\u0007" // Bell Character, not expected from users to store this char in data
   override def transform(dataset: Dataset[_]): DataFrame = {
     require(getInputCol != null, "Input column has to be specified!")
-    val withMarker = dataset.withColumn(s"with_marker_$uid", concat(col(getInputCol), lit(" " + MARKER)))
-    val tokenizer = new Tokenizer()
-      .setInputCol(s"with_marker_$uid")
+    val tokenizer = new RegexTokenizer()
+      .setInputCol(getInputCol())
+      .setMinTokenLength(getMinTokenLength())
       .setOutputCol(s"tokenized_$uid")
-    val tokenized = tokenizer.transform(withMarker)
+      .setPattern(getPattern())
 
     val stopWordsRemover = new StopWordsRemover()
       .setInputCol(tokenizer.getOutputCol)
       .setStopWords(getStopWords())
       .setOutputCol(s"stop_words_removed_$uid")
 
+    val tokenized = tokenizer
+      .transform(dataset)
+      .withColumn(tokenizer.getOutputCol, array_union(col(tokenizer.getOutputCol), lit(Array(""))))
```

Comment:
AFAIK `array_union` was introduced in Spark 2.4. I'm afraid this won't work in older Spark versions.

Reply:
Thanks!

Reply:
Should be fixed, added custom UDF which does the same thing, tested on Spark 2.1 as well